src/libraries/Microsoft.Extensions.Caching.Memory/tests/CapacityTests.cs

```diff
@@ -449,14 +445,23 @@ public void NoCompactionWhenNoMaximumEntriesCountSpecified()
         public void ClearZeroesTheSize()
         {
             var cache = new MemoryCache(new MemoryCacheOptions { SizeLimit = 10 });
-            Assert.Equal(0, cache.Size);
+            AssertCacheSize(0, cache);
 
             cache.Set("key", "value", new MemoryCacheEntryOptions { Size = 5 });
-            Assert.Equal(5, cache.Size);
+            AssertCacheSize(5, cache);
 
             cache.Clear();
-            Assert.Equal(0, cache.Size);
+            AssertCacheSize(0, cache);
             Assert.Equal(0, cache.Count);
         }
+
+        internal static void AssertCacheSize(long size, MemoryCache cache)
+        {
+            // Size is only eventually consistent, so retry a few times
+            RetryHelper.Execute(() =>
+            {
+                Assert.Equal(size, cache.Size);
+            }, maxAttempts: 12, (iteration) => (int)Math.Pow(2, iteration)); // 2ms, 4ms.. 4096 ms. In practice, retries are rarely needed.
```

Comment:
>// 2ms, 4ms.. 4096 ms. In practice, retries are rarely needed. [](http://example.com/codeflow?start=77&length=62)

The max would be around 4 minutes. I think this is reasonable but should we try first with smaller number (max attempt 11)?

Reply:
Is my math wrong? 2^12 is 4,096 msec, so the total overall would be no more than 8 sec?

I can reduce it but in normal circumstances it never retries at all, so it's not hiding a "slow" test.